src/modelfree/visualize/pretty_gym_compete.py

```diff
@@ -0,0 +1,178 @@
+import collections
+import ctypes
+import re
+
+from PIL import Image, ImageDraw, ImageFont
+import gym
+import mujoco_py
+import numpy as np
+
+from modelfree.envs import VICTIM_INDEX
+from modelfree.envs.gym_compete import env_name_to_canonical, game_outcome
+
+VICTIM_OPPONENT_COLORS = {
+    'Victim': (128, 128, 128, 255),
+    'Opponent': (255, 0, 0, 255),
+    'Ties': (0, 0, 0, 255),
+}
+
+
+POLICY_TYPE_COLORS = {
+    'zoo': (128, 128, 128, 255),  # grey
+    'ppo2': (255, 0, 0, 255),  # red
+    'zero': (0, 0, 0, 255),  # black
+    'random': (0, 0, 255, 255),  # blue
+}
+
+
+def body_color(is_victim, agent_type, agent_path):
+    key = 'Victim' if is_victim else 'Opponent'
+    return VICTIM_OPPONENT_COLORS[key]
+
+
+def head_color(is_victim, agent_type, agent_path):
+    return POLICY_TYPE_COLORS[agent_type]
+
+
+GEOM_MAPPINGS = {
+    '*': body_color,
+    # Ant
+    'torso_geom': head_color,
+    # Humanoid
+    'torso1': head_color,
+    'uwaist': head_color,
+    'lwaist': head_color,
+    'head': head_color,
+}
+
+
+def set_geom_rgba(model, value):
+    """Does what model.geom_rgba = ... should do, but doesn't because of a bug in mujoco-py."""
+    val_ptr = np.array(value, dtype=np.float32).ctypes.data_as(ctypes.POINTER(ctypes.c_float))
+    ctypes.memmove(model._wrapped.contents.geom_rgba, val_ptr,
+                   model.ngeom * 4 * ctypes.sizeof(ctypes.c_float))
+
+
+def set_geom_colors(model, patterns):
+    names = [name.decode('utf-8') for name in model.geom_names]
+    patterns = {re.compile(k): tuple(x / 255 for x in v) for k, v in patterns.items()}
+
+    modified = np.array(model.geom_rgba)
+    for row_idx, name in enumerate(names):
+        for pattern, color in patterns.items():
+            if pattern.match(name):
+                modified[row_idx, :] = color
+
+    set_geom_rgba(model, modified)
+
+
+CAMERA_CONFIG = {
+    # From behind
+    'KickAndDefend-v0': {'azimuth': 0, 'distance': 10, 'elevation': -23},
+    # From side, slightly behind (runner always goes forward, never back)
+    'YouShallNotPassHumans-v0': {'azimuth': 110, 'distance': 9, 'elevation': -21},
+    # From side, close up
+    'SumoHumans-v0': {'azimuth': 90, 'distance': 8, 'elevation': -25},
+    'SumoAnts-v0': {'azimuth': 90, 'distance': 10, 'elevation': -25},
+}
+
+
+class PrettyGymCompete(gym.Wrapper):
```

Comment:
Mild preference for this being called something like "AnnotatedGymCompete," since I think that's a closer match to what it's actually doing

Reply:
You don't find the videos pretty? :(

Point taken, `AnnotatedGymCompete` is better.